Source/Model/Model.swift

```diff
@@ -195,33 +214,48 @@ extension Model {
                             let destination = getExecutableFrom(identifier: rIdentifier, codeRequirement: rCodeRequirement)
-                            let appleEvent = AppleEventRule(source: source, destination: destination, value: policy.allowed)
+                            let allowed: Bool = (policy.allowed == true || policy.authorization == TCCPolicyAuthorizationValue.allow)
+                            let appleEvent = AppleEventRule(source: source, destination: destination, value: allowed)
                             executable?.appleEvents.appendIfNew(appleEvent)
                         }
                     } else {
-                        if policy.allowed {
-                            executable?.policy.setValue("Allow", forKey: key)
+                        if policy.authorization == .allow || policy.allowed == true {
+                            executable?.policy.setValue(TCCProfileDisplayValue.allow.rawValue, forKey: key)
+                        } else if policy.authorization == .allowStandardUserToSetSystemService {
+                            executable?.policy.setValue(TCCProfileDisplayValue.allowStandardUsersToApprove.rawValue, forKey: key)
                         } else {
-                            executable?.policy.setValue("Deny", forKey: key)
+                            executable?.policy.setValue(TCCProfileDisplayValue.deny.rawValue, forKey: key)
                         }
                     }
                 }
             }
         }
     }
 
-    func policyFromString(executable: Executable, value: String) -> TCCPolicy? {
-        let allowed: Bool
-        switch value {
-        case "Allow":   allowed = true
-        case "Deny":    allowed = false
-        default:        return nil
-        }
-        return TCCPolicy(identifier: executable.identifier,
+    func policyFromString(executable: Executable, value: String, event: AppleEventRule? = nil) -> TCCPolicy? {
+        var policy = TCCPolicy(identifier: executable.identifier,
                          codeRequirement: executable.codeRequirement,
-                         allowed: allowed)
+                         receiverIdentifier: event?.destination.identifier,
+                         receiverCodeRequirement: event?.destination.codeRequirement)
+        if usingLegacyAllowKey {
+            switch value {
+            case TCCProfileDisplayValue.allow.rawValue:   policy.allowed = true
+            case TCCProfileDisplayValue.deny.rawValue:    policy.allowed = false
+            default:        return nil
+            }
+        } else {
+            switch value {
+            case TCCProfileDisplayValue.allow.rawValue: policy.authorization = .allow
+            case TCCProfileDisplayValue.deny.rawValue: policy.authorization = .deny
+            case TCCProfileDisplayValue.allowStandardUsersToApprove.rawValue: policy.authorization = .allowStandardUserToSetSystemService
+            default:        return nil
```

Comment:
Nit: In this `switch` statement and the one above it, the indentation on these `default` statements is off. Looks like it's just because of carryover from the previous code.

And I personally prefer indenting the code within a `case`, but I'm fine with either.

```swift
case TCCProfileDisplayValue.allow.rawValue:
    policy.authorization = .allow
case TCCProfileDisplayValue.deny.rawValue:
    policy.authorization = .deny
case TCCProfileDisplayValue.allowStandardUsersToApprove.rawValue:
    policy.authorization = .allowStandardUserToSetSystemService
```

Reply:
I agree, but swiftlint complained about it, so I figured it must be the standard? Maybe give it a try and see what you think. I am fine adding a rule to override it it we want.

Reply:
yeah, I can do that